### DISCUSSION TITLE
Video-Downloadhelper Sidebar Always Opens on Mozilla Firefox. No option to turn off the sidebar when opening the program?

### DISCUSSION THREAD

**Discussion by premium9000** [aclap-dev/video-downloadhelper#1192] [Help and Q&A]
### Your question

Hello, so I'm using an Apple M2 Macbook Pro 2023 running Firefox Browser version 129.0.2.  Whenever I open Firefox, the Video-DownloadHelper sidebar always opens on the right side of my screen.

I've searched and searched, but I could not find an option in the menus to stop the video-downloadhelper sidebar from opening when opening Firefox or opening new tabs. 

It's pretty annoying because I always have to close the sidebar manually. 

Is there any way to disable the sidebar when opening Firefox or when making new tabs? or is there a shortcut button combination that I can use to quickly close the sidebar? 

Any help is greatly appreciated because I'm so over having to use my mouse to manually close the sidebar every darn time. 
![Screenshot 2024-08-28 at 3 20 45 PM](https://github.com/user-attachments/assets/dddb0ee4-47de-4d83-be33-63328a10a522)

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"80506933972a4794"}
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "/Volumes/SAB4-10TBSGate/dwhelper (SAB4)",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "/Volumes/SAB4-10TBSGate/10TB NROP/Movies/Organized/00 Waifu Onlyfans Anime EGIRL/eeveedavis Nicole Eevee Davis cosplay waifu white girl 2024"
        }
    },
    "concurrent_downloads_max": 2,
    "show_success_notification_for_icognito": true,
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": true,
        "hide_downloaded": true
    },
    "show_context_menu": true,
    "use_wide_ui": false,
    "use_legacy_ui": false,
    "use_sidebar": false,
    "open_count_store": 1,
    "successfull_dl": 6,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_3569794271018768"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/Users/bryanmiraflorphoto/dwhelper/What makes japanese girls your favorite breeding toy rasiangirls.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "What makes japanese girls your favorite breeding toy rasiangirls.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/Users/bryanmiraflorphoto/dwhelper"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.reddit.com/r/asiangirls4whitecocks/comments/1etjw2a/what_makes_japanese_girls_your_favorite_breeding/#lightbox"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1724169463888
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_6181990456473934"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/Volumes/1TBWD/Movies/naomichaaan sakurasway onlyfans wasian (MSTR)/Naomi-chan uSakuraSway - Reddit3.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Naomi-chan uSakuraSway - Reddit3.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/Volumes/1TBWD/Movies/naomichaaan sakurasway onlyfans wasian (MSTR)"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.reddit.com/user/SakuraSway/"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1724169564222
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_4029347517391331"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/Volumes/SAB4-10TBSGate/10TB NROP/Movies/20240320 organize/l!ng petite asian creampie #37/Bbylingfuckingsonofbitch8-Zh7TYmkC mp4 Bunkr.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Bbylingfuckingsonofbitch8-Zh7TYmkC mp4 Bunkr.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/Volumes/SAB4-10TBSGate/10TB NROP/Movies/20240320 organize/l!ng petite asian creampie #37"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://bunkr.si/v/RTM4bur65F8lK"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1724844071581
                        }
                    }
                ]
            }
        ]
    },
    "record_download_history": true
}

**Comment by GrampaWildWilly** [comment]
![#01](https://github.com/user-attachments/assets/a2942846-7e14-4481-be0a-14bdaf7fb7f3)

With judicious use of F6, Tab, Enter, & Spacebar, you can revert the sidebar to the regular old-fashioned VDH menu permanently without using your mouse.  You need to use the control I indicate in that image, not any other way of closing the sidebar.

**Reply by GrampaWildWilly** [reply]
You might need to scroll the VDH sidebar to see those controls at the bottom.  You also might need to make it wider (drag the edge with the mouse, a function that is not available though the keyboard as far as I know).

**Reply by premium9000** [reply]
I don't think you understand the issue I've been trying to explain.

even though I use that button that you're pointing to with arrows  in your screenshot, sure the sidebar closes. 

But if I open a new firefox tab or if I close Firefox and start the program again, the VDH sidebar automatically opens. 

how do I permanently disable the sidebar from automatically opening when starting up Firefox and not allow VDH to open the sidebar when I make a new tab???

**Reply by Mattmjs** [reply]
I wonder if you tried this if the problem persists or not. Copy your license key and put it in a text editor.
Next click the button to go back to the popup in the toolbar. Then you want to remove the extension from Firefox.
After this the extension should play no part in the browser when you close it and then start the browser again.
What do you see this time around ? still the same or it fixes it.

**Reply by premium9000** [reply]
> I wonder if you tried this if the problem persists or not. Copy your license key and put it in a text editor. Next click the button to go back to the popup in the toolbar. Then you want to remove the extension from Firefox. After this the extension should play no part in the browser when you close it and then start the browser again. What do you see this time around ? still the same or it fixes it.

interesting, so the sidebar still opens when I open a new instance of FireFox.

However, the sidebar does not re-open if I'm creating a new tab. So I guess it semi-fixed the issue. 

I'd still like for an option where the sidebar didn't open at all when opening up Firefox. An option to permanently disable the sidebar from opening.

**Reply by GrampaWildWilly** [reply]
A new Firefox came out today.  130.0.  You might want to be sure you're using that.  I don't know if there's anything fixed in there that affects this issue.  When I open, let's say, the History sidebar, then I close it, I don't subsequently get any sidebar.  There's a couple of ways of closing the History sidebar:

- Ctrl-h
- Click the x at the top of the sidebar
- Drop down the menu at the top of the sidebar & select the Close entry on that menu

Does none of those actions durably get rid of the sidebar?  Failing that, you probably need to do a web search for this problem.  I suspect there's some cryptic about:config setting that is giving you this unwanted behavior.  Cryptic about:config setting.  Now there's a redundant expression if ever there was one.

**Comment by MoDotEu** [comment]
Step 1 : close firefox with only one new tab
![image](https://github.com/user-attachments/assets/cbc00d09-0740-4248-981d-d3dbd96e1ff7)
Step 2: reopen  - of course Video DownloadHelper get open as well
Step 3: do what  GrampaWildWilly suggests, Ctrl-H, Click x, etc...
Step 4: close FireFox again completely
![image](https://github.com/user-attachments/assets/f8d30fbf-01b4-4d42-bd54-e98807920dfa)
Step 5 reOpen FireFox again
![image](https://github.com/user-attachments/assets/b9f971e1-1108-4d56-8fff-5fe26d38222b)
Step 6: remove the Video DownloadHlper completely, accept that this plugin is stronger than you, hope that developers fix this bug, or .. Just go mad!

**Reply by GrampaWildWilly** [reply]
But you're getting a sidebar even when you don't have VDH installed.  That means it's not a VDH bug but a Firefox bug.  I suggest again.  Do a web search & see if this is something anybody else has reported.  Like I said, I suspect there is some about:config setting that is causing this behavior, but you won't find it unless you do a web search.

**Reply by GrampaWildWilly** [reply]
Oops.  Sorry.  That was another user who was getting a sidebar when VDH is not even installed.  Are you seeing that as well?

**Reply by MoDotEu** [reply]
The VDH is the problem here.
When i uninstall VDH, all fine with the Firefox, the sidebar does not appear by starting Firefox.
Reinstall VDH, causes the issue again.
Uninstalling the VDH, will solve the problem. As i mentioned in my previous comment.

**Comment by GrampaWildWilly** [comment]
Even though I am not having this problem, I did a web search using this search key:

firefox bookmarks sidebar always open

Did it not occur to anybody else to do this web search?  That search turned up this:

https://support.mozilla.org/en-US/questions/1444218

Sounds useful to me.  On my system, the file mentioned in that discussion occurs in 2 directories.  I found those directories using a dir /s command to find all instances of the file named in that discussion.  But the date/time stamps & the file sizes on the 2 files are identical.  I suspect you need to perform the advice given in that Mozilla Support discussion on both files.

No, I will not spoonfeed you what it says over there.  Do at least a little bit for yourself & go read it yourself.

**Reply by premium9000** [reply]
nah dude. it's not a bookmark window issue. turning on firefox till this day still opens every time with the VDH sidebar. there's no option to disable it. 

this obviously isn't a one person problem as I see you've been replying to another user with the same issue: 
https://github.com/aclap-dev/video-downloadhelper/discussions/865

**Reply by premium9000** [reply]
they said the sidebar on firefox is a bug that's currently being worked on. but who knows how long until that gets fixed.

https://github.com/aclap-dev/video-downloadhelper/discussions/865#discussioncomment-9812535

**Reply by GrampaWildWilly** [reply]
Did you try deleting the file mentioned in that Mozilla Support issue?  I mean the one I mentioned a couple of posts upthread here.

**Reply by GrampaWildWilly** [reply]
In #840 on June 8, Paul gives a URL to a discussion on one of the Mozilla developer sites.  It looks to me like the bug correction has been included in Firefox.  It's not too clear what version number of Firefox it's been included in, but it's been included in something.  Maybe a dev version.  I don't know.  In the meantime, I still advise that you read the article I referred to above &  delete the file they talk about in there.

**Comment by caddytan** [comment]
This is how I resolve this issue on my Mac.

1. In Firefox, go to Help --> More Troubleshooting Information
2. Scroll down to Profile Folder and click on Show In Finder. It will bring up Finder in the Profiles folder
3. Quit Firefox
4. In the profiles folder, delete xulstore.json
5. start Firefox and sidebar is not started automatically.

**Reply by knallie** [reply]
Thanks caddytan! Your solution worked for me fine:
- recent Firefox version 134.0 on Win10
- VDH version 9.1.0.44

**Reply by GrampaWildWilly** [reply]
@caddytan simply copied the instructions from the support.mozilla.org article I cited back on September 9 upthread.
